parler/fields.py

```diff
@@ -162,3 +166,39 @@ def __set__(self, instance, value):
 
     def __delete__(self, instance):
         raise AttributeError("The 'language_code' attribute cannot be deleted!")
+
+
+class DONOTHING(object):
+    pass
+
+
+if (1, 8) <= django.VERSION < (2, 0):
+    class CompositeOneToOneVirtualField(CompositeOneToOneField):
+        """
+        Class to fix problem with creation repetitive migrations
+        """
+        def deconstruct(self):
+            name, path, args, kwargs = super(CompositeOneToOneVirtualField, self).deconstruct()
+            if 'to_fields' in kwargs:
+                kwargs['to_fields'] = {'master_id': None, 'language_code': None}  # hack: Need always the same dict
+            if "on_delete" in kwargs:
+                kwargs['on_delete'] = DONOTHING  # hack: Need always the same global object with __module__ attr
```

Comment:
What is the purpose of this hack?

Reply:
The hack is need because repitative migrations problem, here
https://github.com/Andrey-Skvortsov/django-composite-foreignkey/blob/91c093c8c81b38a8267557597c6bc0870e4315c3/compositefk/fields.py#L48
they are doing overriding on_delete and Django migrations produced a new migration then do `makemigrations` with the same code even there are no any changes. It's do such because it compare the models's field and found diff in on_delete as it different instances here.
It's already fixed here:
https://github.com/Andrey-Skvortsov/django-composite-foreignkey/blob/91c093c8c81b38a8267557597c6bc0870e4315c3/compositefk/fields.py#L196
but not in PyPi yet, so it needs bumped version in django-composite-foreignkey.